app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt

```diff
@@ -107,34 +119,67 @@ class TopicListAdapter(
     return itemList.size
   }
 
-  private class WelcomeViewHolder(
-    val binding: WelcomeBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
+  private class WelcomeViewHolder(val binding: WelcomeBinding) : RecyclerView.ViewHolder(binding.root) {
     internal fun bind(userAppHistoryViewModel: UserAppHistoryViewModel) {
       binding.viewModel = userAppHistoryViewModel
     }
   }
 
-  private class PromotedStoryViewHolder(
-    val binding: PromotedStoryCardBinding
-  ) : RecyclerView.ViewHolder(binding.root) {
-    internal fun bind(promotedStoryViewModel: PromotedStoryViewModel) {
-      binding.viewModel = promotedStoryViewModel
+  inner class PromotedStoryListViewHolder(val binding: PromotedStoryListBinding) :
+    RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(
+      activity: AppCompatActivity,
+      promotedStoryListViewModel: PromotedStoryListViewModel,
+      promotedStoryList: MutableList<PromotedStoryViewModel>
+    ) {
+      binding.viewModel = promotedStoryListViewModel
+      val promotedStoryAdapter = PromotedStoryListAdapter(promotedStoryList)
+      val horizontalLayoutManager = LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
+      binding.promotedStoryListRecyclerView.apply {
+        layoutManager = horizontalLayoutManager
+        adapter = promotedStoryAdapter
+      }
+
+      val snapHelper = PagerSnapHelper()
+      binding.promotedStoryListRecyclerView.layoutManager = horizontalLayoutManager
+      snapHelper.attachToRecyclerView(binding.promotedStoryListRecyclerView)
+
+      val padding48 = dipToPixels(48)
+      val padding16 = dipToPixels(16)
+      if (promotedStoryList.size > 1) {
```

Comment:
This is a bit hacky and won't work correctly for RTL layouts. Why do we need it?

Reply:
Actually as per @mschanteltc suggestions the implementation should be like this,
If there are more than 1 items, then the left and right margin are uneven, so as to make visual room for next item
![Screenshot_1575270381](https://user-images.githubusercontent.com/9396084/69937318-78daa600-1500-11ea-86de-93d72140821a.png)

But if there is only one item, it will become an issue, so in that case, left/right margin should be same on both sides
![Screenshot_1575270405](https://user-images.githubusercontent.com/9396084/69937363-99a2fb80-1500-11ea-9e76-0a1b9697c5d7.png)

Reply:
Thanks, that makes sense to me. Could we also solve this by setting end padding on each individual item, but set the start padding on the whole recycler view? My thinking is that the padding/margin probably should be the same for each item, and in the case when there is only one item then ensuring the spacing is correct falls on the container itself.

Reply:
We cannot change the padding/margin of recyclerview because all the items in HomeFragment are at a particular margin/padding on left and right but as the promoted story is a carousel, it will use the entire width of screen when learner scrolls, therefore the recyclerview needs to have padding and margin as 0. And the padding/margin should be controlled on items(promoted story) only.

Reply:
Ah. Hmm. I’ll need to think on this to see if there’s a cleaner way to do it. Let’s go with this now.

Reply:
Okay.